packages/react-server-dom-webpack/src/ReactFlightWebpackNodeLoader.js

```diff
@@ -70,19 +70,23 @@ export async function resolve(
       );
     }
   }
-  // We intentionally check the specifier here instead of the resolved file.
-  // This allows package exports to configure non-server aliases that resolve to server files
-  // depending on environment. It's probably a bad idea to export a server file as "main" though.
-  if (specifier.endsWith('.server.js')) {
+  const resolved = defaultResolve(specifier, context, defaultResolve);
```

Comment:
You need to await this, no?

Reply:
Oh. I guess I could.

I tried to `then` it but the real default one was sync. But await should work regardless.

Reply:
You can update the type to be T | Promise<T> so that it would've caught that.